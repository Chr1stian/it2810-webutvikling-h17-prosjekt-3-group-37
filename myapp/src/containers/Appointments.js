import React, { Component } from 'react';
import './../style/App.css';
//import Moment from 'react-moment';
import moment from 'moment';
import DatePicker from 'react-datepicker';

import 'react-datepicker/dist/react-datepicker.css';

export class Appointments extends Component {
  constructor(props){
    super(props);
    this.createAppointment = this.createAppointment.bind(this);
    this.state = {
      startDate: moment(),
      hour: '12',
      minute: '30'
    };
    this.handleTimeChange = this.handleTimeChange.bind(this);
    this.handleMinuteChange = this.handleMinuteChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }
  handleTimeChange(event) {
    this.setState({hour: event.target.value});
  }

  handleMinuteChange(event) {
    this.setState({minute: event.target.value});
  }

  handleSubmit(event) {
    alert('Your favorite time is: ' + this.state.hour + this.state.minute);
    event.preventDefault();
  }


  handleChange(date) {
    this.setState({
      startDate: date
    });
  }
  createAppointment(){
    document.getElementById('change').innerHTML = "HATE";
  }
  render() {
    return (
      <div className="Appointments">
        <header className="Appointments-header">
          <h1 className="Appointments-title">Add or see your Appointments</h1>
        </header>

        <button onClick={this.createAppointment}>Create Appointment</button>



        <p className="Appointments-intro">
          To get started, click an Appointment og create a new one.
        </p>
          <table>
          <tbody>
            <tr>
              <th id="change">Date</th>
              <th>Title</th>
              <th>Time</th>
              <th>Place</th>
            </tr>
            <tr>
              <td>Today</td>
              <td>Møte med Petter</td>
              <td>12:30 - 13:30</td>
              <td>NTNU, Gløshaugen</td>
            </tr>
            <tr>
              <td>12. Oktober</td>
              <td>Levere Øving</td>
              <td>10:00 - 10:15</td>
              <td>Sentralbygg 2</td>
            </tr>
            <tr>
              <td>12. Oktober</td>
              <td>Spille Smash</td>
              <td>10:30 - 11:00</td>
              <td>Onlinekontoret</td>
            </tr>
            <tr>
              <td>13. Oktober</td>
              <td>Jobbe med IR</td>
              <td>10:15 - 14:00</td>
              <td>P15 Moki</td>
            </tr>
          </tbody>
          </table>

          <div className="Create-Appointment">
            <h1 className="Appointments-title">Create new appointment</h1>
            <form onSubmit={this.handleSubmit}>
              <label>
                Title:
                <input type="text" name="title" />
              </label>
              <DatePicker
              dateFormat="DD/MM/YYYY"
              selected={this.state.startDate}
              onChange={this.handleChange} />
              <select value={this.state.hour} onChange={this.handleTimeChange}>
                <option value="00">00</option>
                <option value="01">01</option>
                <option value="02">02</option>
                <option value="03">03</option>
                <option value="04">04</option>
                <option value="05">05</option>
                <option value="06">06</option>
                <option value="07">07</option>
                <option value="08">08</option>
                <option value="09">09</option>
                <option value="10">10</option>
                <option value="11">11</option>
                <option value="12">12</option>
                <option value="13">13</option>
                <option value="14">14</option>
                <option value="15">15</option>
                <option value="16">16</option>
                <option value="17">17</option>
                <option value="18">18</option>
                <option value="19">19</option>
                <option value="20">20</option>
                <option value="21">21</option>
                <option value="22">22</option>
                <option value="23">23</option>
              </select>
              <select value={this.state.minute} onChange={this.handleMinuteChange}>
                <option value="15">15</option>
                <option value="30">30</option>
                <option value="45">45</option>
              </select>
              <label>
                Place:
                <input type="text" name="place" />
              </label>

              <input type="submit" value="Submit" />
            </form>

            
          </div>

      </div>
    );
  }
}

export default Appointments;
