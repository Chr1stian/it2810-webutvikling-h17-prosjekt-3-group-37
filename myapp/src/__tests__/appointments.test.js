import React from "react";
//Import Enzyme with shallow to test component as a unit
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
Enzyme.configure({ adapter: new Adapter() });
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
//Importing component that will be tested
import Appointments from "./../containers/Appointments.js";

//Since localstorage is saved in the browser this mocks the data 
global.window = {};
import localStorage from "mock-local-storage";
window.localStorage = global.localStorage;

test("Component renders as expected", () => {
  const component = shallow(<Appointments />);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});