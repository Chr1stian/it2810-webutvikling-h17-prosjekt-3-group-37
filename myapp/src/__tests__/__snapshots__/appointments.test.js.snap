// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders as expected 1`] = `
<div
  className="Appointment"
>
  <Card
    className="SeeAppointment"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
  >
    <CardTitle
      title="Your Appointments"
    />
    <Divider
      inset={false}
    />
    <Table
      allRowsSelected={false}
      className="AppointmentTable"
      fixedFooter={true}
      fixedHeader={true}
      height="inherit"
      multiSelectable={false}
      selectable={true}
    >
      <TableHeader
        adjustForCheckbox={false}
        displaySelectAll={false}
        enableSelectAll={true}
        selectAllSelected={false}
      >
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          striped={false}
        >
          <TableHeaderColumn
            style={
              Object {
                "fontSize": "20px",
                "width": 100,
              }
            }
          >
            Date
          </TableHeaderColumn>
          <TableHeaderColumn
            style={
              Object {
                "fontSize": "20px",
                "width": 150,
              }
            }
          >
            Title
          </TableHeaderColumn>
          <TableHeaderColumn
            style={
              Object {
                "fontSize": "20px",
                "width": 100,
              }
            }
          >
            Time
          </TableHeaderColumn>
          <TableHeaderColumn
            style={
              Object {
                "fontSize": "20px",
                "width": 150,
              }
            }
          >
            Place
          </TableHeaderColumn>
          <TableHeaderColumn>
            <RaisedButton
              disabled={false}
              fullWidth={false}
              labelPosition="after"
              primary={true}
              secondary={false}
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              Delete
            </RaisedButton>
          </TableHeaderColumn>
        </TableRow>
      </TableHeader>
      <TableBody
        allRowsSelected={false}
        deselectOnClickaway={true}
        displayRowCheckbox={true}
        multiSelectable={false}
        preScanRows={true}
        selectable={true}
        style={Object {}}
      />
    </Table>
  </Card>
  <Card
    className="CreateAppointment"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
  >
    <CardTitle
      title="Create Appointment"
    />
    <Divider
      inset={false}
    />
    <div
      className="FormFields"
    >
      <TextField
        disabled={false}
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="Enter title"
        id="titleText"
        multiLine={false}
        rows={1}
        type="text"
        underlineShow={true}
      />
      <DatePicker
        DateTimeFormat={[Function]}
        autoOk={false}
        container="dialog"
        disableYearSelection={false}
        disabled={false}
        firstDayOfWeek={1}
        hideCalendarDate={false}
        hintText="Select date"
        id="dateValue"
        locale="en-GB"
        minDate={2017-10-18T10:31:31.280Z}
        openToYearSelection={false}
        style={Object {}}
      />
      <TimePicker
        autoOk={false}
        cancelLabel="Cancel"
        defaultTime={null}
        disabled={false}
        format="24hr"
        hintText="Select start-time"
        id="fromTime"
        minutesStep={15}
        okLabel="OK"
        pedantic={false}
        style={Object {}}
        value={null}
      />
       
      <TimePicker
        autoOk={false}
        cancelLabel="Cancel"
        defaultTime={null}
        disabled={false}
        format="24hr"
        hintText="Select end-time"
        id="toTime"
        minutesStep={15}
        okLabel="OK"
        pedantic={false}
        style={Object {}}
        value={null}
      />
      <TextField
        disabled={false}
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="Enter place/address"
        id="placeText"
        multiLine={false}
        rows={1}
        type="text"
        underlineShow={true}
      />
      <RaisedButton
        disabled={false}
        fullWidth={false}
        id="addAppointment"
        labelPosition="after"
        onClick={[Function]}
        primary={true}
        secondary={false}
      >
        Add Appointment
      </RaisedButton>
    </div>
  </Card>
</div>
`;
